sm/plugins/tedge_docker_plugin/tedge_docker_plugin.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+unsupported_args_check() {
+    if ! [ -z $1 ]; then
+        echo "Unsupported arguments: $@"
+        exit 1
+    fi
+}
+
+extract_image_tag_from_args() {
+    IMAGE_NAME="$1"
+    if [ -z "$IMAGE_NAME" ]; then
+        echo "Image name is a mandatory argument"
+        exit 1
+    fi
+    shift   # Pop image name from args list
+    IMAGE_TAG=$IMAGE_NAME
+    
+    if ! [ -z $1 ]; then
+        case "$1" in 
+            --module-version)
+                IMAGE_VERSION="$2"
+                IMAGE_TAG=$IMAGE_NAME:$IMAGE_VERSION
+                shift 2  # Pop --version and the version value from the args list
+                ;;
+            *)
+                echo "Unsupported argument option: $1"
+                exit 1
+                ;;
+        esac
+    fi
+    
+    unsupported_args_check $@
+
+}
+
+set -e
```

Comment:
Setting `-e` will make the script to exit with status code inherited from the command which failed and this is not behaviour agent would expect.

Reply:
Good point.

Since our expectation for a plugin is to exit with status code 2, the simple is to add `|| exit 2` after each docker command.

For instance:

```shell
docker run -d $IMAGE_TAG || exit 2
```

Reply:
This line has still to be removed.